src/main/java/sparksoniq/jsoniq/compiler/translator/expr/flowr/ForClauseVar.java

```diff
@@ -45,6 +46,12 @@ public ForClauseVar(
         super(FLWOR_CLAUSES.FOR_VAR, varRef, seq, expression, metadataFromContext);
         this.allowEmpty = emptyFlag;
         this.positionalVariableReference = atVarRef;
+        if (seq != null)
+            this.asSequence = new FlworVarSequenceType(
+                    seq.getSequence().getItemType().getType(),
```

Comment:
I guess this will then use the static variable introduced in the other PR, correct?

Reply:
Currently asSequence property in FlworVarDecl is null by default and all the places that need it, they first check for nullity. We could stick to this, and then also this part of the code is fine at its current state, or we set the default for this property to a new FlworVarSequenceType with the _mostGeneralSequenceType_ introduced in the other PR, and change all checks for nullity with checks for this new most-general FlworVarSequenceType